src/python_testing/TC_AccessChecker.py

```diff
@@ -0,0 +1,207 @@
+    READ = auto()
+    WRITE = auto()
+
+
+def operation_allowed(spec_requires: Clusters.AccessControl.Enums.AccessControlEntryPrivilegeEnum,
+                      acl_set_to: Optional[Clusters.AccessControl.Enums.AccessControlEntryPrivilegeEnum]) -> bool:
+    ''' Determines if the action is allowed on the device based on the spec_requirements and the current ACL privilege granted.
+
+        The spec parsing uses kUnknownEnumValue to indicate that NO access is allowed for this attribute
+        or command (ex. command with no write access).
+        ACL uses None to indicate that no access has been granted to this controller.
+        In both of these cases, the action is disallowed. In all other cases, access is allowed if the ACL
+        grants a privilege at or above the privilege required in the spec.
+    '''
+    if spec_requires == Clusters.AccessControl.Enums.AccessControlEntryPrivilegeEnum.kUnknownEnumValue or acl_set_to is None:
+        return False
+    return spec_requires <= acl_set_to
+
+
+def checkable_attributes(cluster_id, cluster, xml_cluster) -> list[uint]:
+    all_attrs = cluster[GlobalAttributeIds.ATTRIBUTE_LIST_ID]
+
+    def known_cluster_attribute(attribute_id) -> bool:
+        ''' Returns true if this is a non-manufacturer specific attribute that has information in the XML and has python codegen data'''
+        return attribute_id <= 0xFFFF and attribute_id in xml_cluster.attributes and attribute_id in Clusters.ClusterObjects.ALL_ATTRIBUTES[cluster_id]
+    return [x for x in all_attrs if known_cluster_attribute(x)]
+
+
+class AccessChecker(MatterBaseTest, BasicCompositionTests):
+    @async_test_body
+    async def setup_class(self):
+        # TODO: Make this into a proper default in the class so we're not overriding the command lines
+        self.user_params["use_pase_only"] = False
+        super().setup_class()
+        await self.setup_class_helper()
+        self.xml_clusters, self.problems = build_xml_clusters()
+        acl_attr = Clusters.AccessControl.Attributes.Acl
+        self.default_acl = await self.read_single_attribute_check_success(cluster=Clusters.AccessControl, attribute=acl_attr)
+        self._record_errors()
+        # We need to run this test from two controllers so we can test access to the ACL cluster while retaining access to the ACL cluster
+        fabric_admin = self.certificate_authority_manager.activeCaList[0].adminList[0]
+        self.TH2_nodeid = self.matter_test_config.controller_node_id + 1
+        self.TH2 = fabric_admin.NewController(nodeId=self.TH2_nodeid)
+
+    @async_test_body
+    async def setup_test(self):
+        self.success = True
+
+    @async_test_body
+    async def teardown_test(self):
+        await self._cleanup_acl(default_acl=self.default_acl)
+
+    async def _setup_acl(self, default_acl: list[Clusters.AccessControl.Structs.AccessControlEntryStruct], privilege: Optional[Clusters.AccessControl.Enums.AccessControlEntryPrivilegeEnum]):
```

Comment:
nit: default_acl both on here and cleanup seems to be `self.default_acl` always. Would it make sense to remove this parameter?

Reply:
Updated in latest commit.